src/components/ui/ImageGallery/ImageGallerySelector.tsx

```diff
@@ -0,0 +1,130 @@
+import type { HTMLAttributes, PropsWithChildren } from 'react'
+import React, { useMemo } from 'react'
+import { useSlider, IconButton } from '@faststore/ui'
+
+import { ForwardArrowIcon, BackwardArrowIcon } from './Icons'
+
+interface Props extends HTMLAttributes<HTMLDivElement> {
+  itemsPerPage: number
+}
+
+const createTransformValues = (totalItems: number) => {
+  if (!totalItems) {
+    return {}
+  }
+
+  const transformMap: Record<number, number> = {}
+  const slideWidth = 100 / totalItems
+
+  for (let idx = 0; idx < totalItems; ++idx) {
+    const transformValue = -(slideWidth * idx)
+
+    transformMap[idx] = transformValue
+  }
+
+  return transformMap
+}
+
+function ImageGallerySelector({
+  itemsPerPage,
+  children,
+  ...otherProps
+}: PropsWithChildren<Props>) {
+  const elements = React.Children.toArray(children)
+  const elementCount = elements.length
+
+  const transformValues = useMemo(
+    () => createTransformValues(elementCount),
+    [elementCount]
+  )
+
+  const { handlers, slide, sliderState, sliderDispatch } = useSlider({
+    totalItems: elementCount,
+    itemsPerPage,
+    infiniteMode: false,
+  })
+
+  const slidingTransition = `transform 400ms`
+
+  if (!elements || !elementCount) {
+    return null
+  }
+
+  return (
+    <section>
+      <IconButton
+        aria-label="slide image selector backward"
+        icon={<BackwardArrowIcon color="#323845" />}
+        onClick={() => {
+          if (sliderState.sliding) {
+            return
+          }
+
+          slide('previous', sliderDispatch)
+        }}
+      />
+      <IconButton
+        aria-label="slide image selector forward"
+        icon={<ForwardArrowIcon color="#323845" />}
+        onClick={() => {
+          if (sliderState.sliding) {
+            return
+          }
+
+          slide('next', sliderDispatch)
+        }}
+      />
+
+      <div {...otherProps} {...handlers}>
+        <div
+          data-carousel-track
+          style={{
+            display: 'flex',
+            transition: sliderState.sliding ? slidingTransition : undefined,
+            width: `${(elementCount * 100) / itemsPerPage}%`,
+            transform: `translate3d(${
```

Comment:
🆒 !

Reply:
Credits to @georgebrindeiro

Reply:
Something interesting, why not calculate only the current transformValue instead of all values? I am asking because I wanna remove the useMemo, if it's possible.

Reply:
Good call. I've removed the `useMemo` here: https://github.com/vtex-sites/base.store/pull/142/commits/d244c4dec6b48a0c4a97f8f7379f88285bd8739d